src/astro/databases/postgres.py

```diff
@@ -26,3 +29,68 @@ def hook(self):
     def default_metadata(self) -> Metadata:
         # TODO: comment why
         return Metadata(database=self.hook.schema)
+
+    def merge_table(
+        self,
+        source_table: Table,
+        target_table: Table,
+        conflict_strategy: DBMergeConflictStrategy = "ignore",
+        merge_cols: Optional[List[str]] = None,
+        source_tables_cols: Optional[List[str]] = None,
+        target_tables_cols: Optional[List[str]] = None,
+    ) -> None:
+        """Merge two tables based on merge keys
+
+        :param source_table: Contains the rows to be appended to the target_table
+        :param target_table: Contains the destination table in which the rows will be appended
+        :param conflict_strategy: Action that needs to be taken in case there is a conflict
+        :param merge_cols: List of cols that are checked for uniqueness while merging,
+         they will be the unique post merge
+        :param source_tables_cols: List of columns name in source table that will be used in merging
+        :param target_tables_cols: List of columns name in target table that will be used in merging
+        """
+        source_tables_cols = source_tables_cols or []
+        target_tables_cols = target_tables_cols or []
+        merge_cols = merge_cols or []
+
+        if len(source_tables_cols) != len(target_tables_cols):
+            raise ValueError(
+                "Params 'source_tables_cols' and  'target_tables_cols' should contain same number of cols."
+            )
+
+        def identifier_args(table):
+            schema = table.metadata.schema
+            return (schema, table.name) if schema else (table.name,)
+
+        statement = "INSERT INTO {main_table} ({target_columns}) SELECT {append_columns} FROM {append_table}"
+
+        if len(merge_cols) > 0:
+            if conflict_strategy == "ignore":
+                statement += " ON CONFLICT ({merge_keys}) DO NOTHING"
+            elif conflict_strategy == "update":
+                statement += (
+                    " ON CONFLICT ({merge_keys}) DO UPDATE SET {update_statements}"
+                )
+
+        source_tables_cols = [postgres_sql.Identifier(c) for c in source_tables_cols]
```

Comment:
We could have a dedicated method to return table columns in the Psyco format: `_psycopg_table_columns`.
Smaller functions can be easily unit-tested.